source/simulator/core.cc

```diff
@@ -2241,6 +2256,297 @@ namespace aspect
           break;
         }
 
+        case NonlinearSolver::Newton_Stokes:
+        {
+          if (parameters.free_surface_enabled)
+            free_surface->execute ();
+
+          double initial_temperature_residual = 0;
+          double initial_stokes_residual      = 0;
+          std::vector<double> initial_composition_residual (parameters.n_compositional_fields,0);
+
+          //parameters.nonlinear_iteration_number = 0;
```

Comment:
What about this line? I think your scheme needs to set the nonlinear iteration number somehow. Please take a look how the other schemes handle this.